sdk/monitor/monitor-opentelemetry-exporter/src/platform/nodejs/baseSender.ts

```diff
@@ -106,42 +147,77 @@ export abstract class BaseSender {
           diag.info(result);
           const breezeResponse = JSON.parse(result) as BreezeResponse;
           const filteredEnvelopes: Envelope[] = [];
-          // If we have a partial success, count the succeeded envelopes
-          if (breezeResponse.itemsAccepted > 0 && statusCode === 206) {
-            this.networkStatsbeatMetrics?.countSuccess(duration);
-          }
+          // Create a list of successful envelopes by filtering out the failed ones for customer statsbeat
+          const successfulEnvelopes: Envelope[] = [...envelopes];
+
           // Figure out if we need to either retry or count failures
           if (breezeResponse.errors) {
             breezeResponse.errors.forEach((error) => {
+              // Mark as undefined so we don't process them in countSuccessfulEnvelopes
+              successfulEnvelopes[error.index] = undefined as unknown as Envelope;
+
+              // Add to retry list if status code is retriable
               if (error.statusCode && isRetriable(error.statusCode)) {
                 filteredEnvelopes.push(envelopes[error.index]);
               }
             });
           }
+
+          // If we have a partial success, count the succeeded envelopes
+          if (breezeResponse.itemsAccepted > 0) {
+            this.networkStatsbeatMetrics?.countSuccess(duration);
+            // Count only the successful envelopes (non-undefined)
+            if (!this.isStatsbeatSender) {
+              this.countSuccessfulEnvelopes(successfulEnvelopes.filter(Boolean));
+            }
+            // For network statsbeat we just care that some envelopes were successful
+            this.networkStatsbeatMetrics?.countSuccess(duration);
+          }
           if (filteredEnvelopes.length > 0) {
             this.networkStatsbeatMetrics?.countRetry(statusCode);
+            this.customerStatsbeatMetrics?.countRetryItems(
+              filteredEnvelopes.length,
+              RetryCode.RETRYABLE_STATUS_CODE,
+            );
             // calls resultCallback(ExportResult) based on result of persister.push
             return await this.persist(filteredEnvelopes);
           }
           // Failed -- not retriable
           this.networkStatsbeatMetrics?.countFailure(duration, statusCode);
+          // Count dropped items for customer statsbeat for non-retriable status codes
+          this.customerStatsbeatMetrics?.countDroppedItems(
+            envelopes.length - successfulEnvelopes.filter(Boolean).length,
+            DropCode.NON_RETRYABLE_STATUS_CODE,
```

Comment:
This should not be the literal string "NON_RETRYABLE_STATUS_CODE". It should be the actual status code that customer received that is non-retryable.

From spec: 
`*NON_RETRYABLE_STATUS_CODE will be the actual value of the non-retryable status code that was returned.`

Reply:
Updated all instances of NON_RETRYABLE_STATUS_CODE and RETRYABLE_STATUS_CODE to actually send the code.